packages/hoprd/src/identity.ts

```diff
@@ -67,27 +81,40 @@ async function storeIdentity(path: string, id: Uint8Array) {
   fs.writeFileSync(resolve(path), id)
 }
 
-async function createIdentity(idPath: string, password: string, useWeakCrypto = false) {
-  const peerId = await PeerId.create({ keyType: 'secp256k1' })
+async function createIdentity(idPath: string, password: string, useWeakCrypto = false, privateKey?: Uint8Array) {
+  const peerId = privateKey ? privKeyToPeerId(privateKey) : await PeerId.create({ keyType: 'secp256k1' })
```

Comment:
Ternary operator containing promises and non-promises have shown to be sometimes dangerous

Reply:
Why? Do you have an example?

Reply:
Currently, I don't have a good example that breaks but I have seen a few weird things when mixing boolean arithmetic with promise logic. Haven't tested if that got better with Node.js 14